### ISSUE TITLE
training for multiple steps

### ISSUE THREAD

**Issue by danielhomola** [sid-sundrani/ppo_lightning#2]
Thanks for the great work! 

I was curious if you heard back from the PT Lightning crew about how to implement the multi-step update that OpenAI does on the collected trajectories? Right now you're only taking a single step which is arguably sample inefficient, although it seems to work fine for you.

Fig 54 [here](https://arxiv.org/pdf/2006.05990.pdf) seems to suggest that that's a crucial step to getting PPO right though.. 

Thanks,
d

**Comment by sidhantls** [comment]
That is true. This repo, currently, just does single step and does not take advantage of PPO's data efficiency of running multiple steps of GD. Thanks for sharing this.  

I did recently implement the multi-step update that OpenAI does. It just required some manipulation with `configure_optimizers`. I'll push the update soon. EDIT: I just updated the repo with multi-step optimization 

With multiple gradient steps (orange) vs single step (blue, current implementation) same params on Cartpole
![image](https://user-images.githubusercontent.com/19412334/103309602-88962800-4a3b-11eb-90d3-339b0a58fd09.png)

**Comment by danielhomola** [comment]
That looks great. Thanks so much for adding this!

**Comment by danielhomola** [comment]
So [this change](https://github.com/sid-sundrani/ppo_lightning/blob/master/ppo_model.py#L278) will make a new instance of optimizer for each epoch. Did the Lightning guys recommend this workaround? It kind of feels hacky.. Also it prohibits using learning rate schedulers.

**Comment by sidhantls** [comment]
It is not making a new instance of the optimizer for each epoch. It's making one instance of the optimizer and there are multiple references of it in the list for each iteration of optimization

I came up with this particular workaround. You'd have to hack it further to include LR schedulers. Let's see, I'll work on it and I'll be sure to update this approach to better fit into lightning :)

**Comment by danielhomola** [comment]
Ah sure, fair enough it's just the reference! I like this a lot actually :) Thanks again!

**Comment by sidhantls** [comment]
FYI, I just updated the method of doing this. It should now allow you to use schedulers the standard/built-in way in Lightning

**Comment by danielhomola** [comment]
awesome! thanks
